rust_snuba/rust_arroyo/src/processing/mod.rs

```diff
@@ -71,7 +72,7 @@ pub struct StreamProcessor<'a, TPayload: Clone> {
     consumer: Box<dyn Consumer<'a, TPayload> + 'a>,
     strategies: Arc<Mutex<Strategies<TPayload>>>,
     message: Option<Message<TPayload>>,
-    shutdown_requested: bool,
+    pub shutdown_requested: Arc<AtomicBool>,
```

Comment:
I wasn't able to get `processor.signal_shutdown()` to work since StreamProcessor is not thread safe